codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -679,29 +617,18 @@ final void takeChild(DefaultStream child, boolean exclusive, List<ParentChangedE
                 events.add(new ParentChangedEvent(child, oldParent));
                 notifyParentChanging(child, this);
                 child.parent = this;
-                // We need the removal operation to happen first so the prioritizableForTree for the old parent to root
-                // path is updated with the correct child.prioritizableForTree() value. Note that the removal operation
-                // may not be successful and may return null. This is because when an exclusive dependency is processed
-                // the children are removed in a previous recursive call but the child's parent link is updated here.
-                if (oldParent != null && oldParent.children.remove(child.id()) != null &&
-                        !child.isDescendantOf(oldParent)) {
-                    oldParent.decrementPrioritizableForTree(child.prioritizableForTree());
-                    if (oldParent.prioritizableForTree() == 0) {
-                        // There are a few risks with immediately removing nodes from the priority tree:
-                        // 1. We are removing nodes while we are potentially shifting the tree. There are no
-                        // concrete cases known but is risky because it could invalidate the data structure.
-                        // 2. We are notifying listeners of the removal while the tree is in flux. Currently the
-                        // codec listeners make no assumptions about priority tree structure when being notified.
-                        removeStream(oldParent, null);
-                    }
+                // Note that the removal operation may not be successful and may return null. This is because when an
+                // exclusive dependency is processed the children are removed in a previous recursive call but the
+                // child's parent link is updated here.
+                if (oldParent != null) {
+                    oldParent.children.remove(child.id());
```

Comment:
This is part of the tree restructure operation and is not removing from any other collection (e.g. active stream or stream map). This is just a subset of what was done above ... what was done above was necessary because a node that was closed could be restructured to be a leaf node ... its `prioritizableForTree` go to 0 ... and then we should remove it (closed nodes should not exist as leaf nodes). That operation was a bit more "risky" as indicated in the comments above ... but didn't lead to any concrete issues.

I think we are OK but did you have a specific case in mind?